src/utils/utils.js

```diff
@@ -278,14 +278,20 @@ export function checkJsonConditional(component, json, row, data, form, onError)
  * @returns {boolean}
  */
 export function checkCondition(component, row, data, form, instance) {
-  if (component.customConditional) {
-    return checkCustomConditional(component, component.customConditional, row, data, form, 'show', true, instance);
-  }
-  else if (component.conditional && component.conditional.when) {
-    return checkSimpleConditional(component, component.conditional, row, data);
+  const { customConditional, conditional } = component;
+  if (customConditional) {
+    return checkCustomConditional(component, customConditional, row, data, form, 'show', true, instance);
+  }
+  else if (conditional && conditional.when) {
+    if (instance?.parent && conditional.when.startsWith(instance.parent.path)) {
```

Comment:
Would this break if the parent was a panel called "Child" and there was a field inside of it called "Child First Name"?